### PR TITLE
Extract docker-compose.yml for mocks and domain-xample

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

All docker-compose containers were specified in a single monolithic file.

## How does this fix it?
Split and extract docker-compose.yml for `mocks` and `domain-xample`.
This will reduce the need for compose profiles.

## How to test this PR
Run VRO locally:
```
./gradlew :app:dockerComposeUp
# Once the above containers load completely, run:
./gradlew :mocks:dockerComposeUp
./gradlew :domain-xample:dockerComposeUp
```

### CODE REVIEW

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4885695201)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4886892440)

File: .github/workflows/end2end-test.yml [hunk 1]
```diff
@@ -43,14 +43,17 @@ jobs:
...
           export COMPOSE_PROFILES="v2,v2-mocks,pdfgen,lh"
           export -p | sed 's/declare -x //'
+
+          ./gradlew :mocks:dockerComposeUp
+          sleep 20
+          docker ps
+
           ./gradlew :app:dockerComposeUp
+          sleep 20
+          docker ps
```
Comment (yoomlam) [inline .github/workflows/end2end-test.yml:53]: Add a few `docker ps` for diagnostics.

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4887763859)

File: domain-xample/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+version: '3.9'
```
Comment (msnwatson) [inline domain-xample/docker-compose.yml:1]: Should this file and the other be nested in a src/docker dir to match with other uses in the repo?

Comment (yoomlam) [reply domain-xample/docker-compose.yml:1]: I considered that but I decided that the `src/docker` folder will be for building the Docker image of subprojects, while the `docker-compose.yml` belongs in the `domain-*` folder.  I don't anticipate more than one `docker-compose.yml` per domain. I plan to move `app/src/docker/docker-compose.yml` to `app` or the root folder. Open to feedback.

**Pull request merged**